src/main/java/org/ow2/proactive/connector/iaas/service/InfrastructureService.java

```diff
@@ -52,13 +59,16 @@ public Infrastructure registerInfrastructure(Infrastructure infrastructure) {
         return infrastructure;
     }
 
-    public void deleteInfrastructure(Infrastructure infrastructure) {
+    public void deleteInfrastructure(Infrastructure infrastructure, Boolean deleteInstances) {
+        if (deleteInstances) {
```

Comment:
Separate this in another method :  
 deleteInfrastructureAndItsInstances(Infrastructure infrastructure)